src/core/dynamicalsystem_interface.jl

```diff
@@ -241,6 +242,19 @@ but a developer may care.
 """
 SciMLBase.isinplace(ds::DynamicalSystem) = errormsg(ds)
 
+"""
+    successful_step(ds::DynamicalSystem) -> true/false
+
+Return `true` if the last `step!` call to `ds` was successful, `false` otherwise.
+For continuous time systems this uses DifferentialEquations.jl error checking,
+for discrete time it checks if any variable is `Inf` or `NaN`.
+
+"""
+successful_step(ds::DynamicalSystem) = nothing
```

Comment:
```suggestion
successful_step(ds::DynamicalSystem) = errormsg(ds)
```

So that it correctly shows the error instead of being silent.